agent/structs/structs.go

```diff
@@ -452,7 +452,7 @@ func (r *RegisterRequest) ChangesNode(node *Node) bool {
 
 	// Check if any of the node-level fields are being changed.
 	if r.ID != node.ID ||
-		r.Node != node.Node ||
+		!strings.EqualFold(r.Node, node.Node) ||
```

Comment:
Going down the testing rabbit hole a bit but we should probably have tests for this behavior modifications in this file as well.

Reply:
done